src/GafferScene/Light.cpp

```diff
@@ -64,8 +62,12 @@ Light::Light( const std::string &name )
 	storeIndexOfNextChild( g_firstPlugIndex );
 	addChild( new Plug( "parameters" ) );
 	addChild( new BoolPlug( "defaultLight", Gaffer::Plug::Direction::In, true ) );
-	addChild( new FloatPlug( "visualiserScale", Gaffer::Plug::Direction::In, 1.0 ) );
-	addChild( new BoolPlug( "visualiserShaded", Gaffer::Plug::Direction::In, true ) );
+
+	Gaffer::CompoundDataPlug *visualiserAttr = new CompoundDataPlug( "visualiserAttributes" );
+	visualiserAttr->addChild( new Gaffer::NameValuePlug( "gl:visualiser:ornamentScale", new IECore::FloatData( 1.0f ), false, "ornamentScale" ) );
```

Comment:
There's a version of the NameValuePlug constructor that takes a ValuePlug instead of a DataPtr. If we use that we can set sensible minimum values for scale and max resolution - currently they can both go negative. Could maybe also have a sensible upper limit for the max resolution?

Reply:
Limits added in  https://github.com/GafferHQ/gaffer/pull/3519/commits/3b334830f74592fb556b71f91e8d68afba5a9bd1, also for other attributes as discussed offline.